### PR TITLE
ci: if the test fails, github action should fail.

### PR DESCRIPTION
**Pull request by amondnet**
Describe the changes proposed in this Pull Request.

If the test fails, github action should fail.
https://github.com/mobxjs/mobx.dart/pull/784 has failed tests, but [check](https://github.com/mobxjs/mobx.dart/runs/6815659674?check_suite_focus=true) succeeded.

---
## Pull Request Checklist

- [x] If the changes are being made to code, ensure the **version in `pubspec.yaml`** is updated. 
- [x] Increment the **`major`/`minor`/`patch`/`patch-count`**, depending on the complexity of change
- [x] Add the necessary **unit tests** to ensure the coverage does not drop
- [x] Update the **Changelog** to include all changes made in this PR
- [x] Run the **`set:versions` command** using `npm` or `yarn`. You can find this command in the `package.json` file in the root directory
- [x] Include the **necessary reviewers** for the PR
- [x] Update the docs if there are any API changes or additions to functionality

### CODE REVIEW

**Comment by amondnet** [comment]
https://github.com/mobxjs/mobx.dart/runs/6959595920?check_suite_focus=true

**Review by fzyzcjy — COMMENTED** [review]
LGTM

**Comment by pavanpodila** [comment]
Is this working @amondnet ? There are lot of deleted lines in `coverage.sh`. Not sure if that is running the tests as expected, including the collection of coverage

**Comment by amondnet** [comment]
@pavanpodila Like the `flutter test`, the `dart test` has a `coverage` argument and .

https://github.com/mobxjs/mobx.dart/blob/c963f6cba2fdf23d7ba42ce44829152adfcd105d/.github/workflows/build.yml#L62

https://pub.dev/packages/test#collecting-code-coverage
https://github.com/dart-lang/coverage#running-tests-with-coverage

https://github.com/dart-lang/test/pull/1088
https://github.com/dart-lang/test/pull/1600


I tested mobx, 2.0.7. 
`git checkout mobx-2.0.7 `

1. use `dart test`
  ```
  dart run test --coverage=./coverage test/all_tests.dart
  dart run coverage:format_coverage \
    --lcov \
    --in=coverage/coverage.json \
    --out=coverage/lcov.info \
    --packages=.dart_tool/package_config.json \
    --report-on=lib
  ```
  https://gist.github.com/amondnet/be02b37c58cae3a1740fb9d1a6747a3f#file-dart-lcov
  2. use `tool/coverage.sh`
  https://gist.github.com/amondnet/be02b37c58cae3a1740fb9d1a6747a3f#file-tool-lcov
  
text compare

스크린샷 2022-06-27 오후 1 00 05


https://gist.github.com/amondnet/be02b37c58cae3a1740fb9d1a6747a3f

File: mobx_codegen/lib/src/type_names.dart [hunk 1]
```diff
@@ -35,6 +35,7 @@ class LibraryScopedNameFinder {
     // Reverse each import's export namespace so we can map elements to their
     // library-local names. Note that the definedNames include a prefix if there
     // is one.
+    // ignore: deprecated_member_use
```
Comment (amondnet) [inline mobx_codegen/lib/src/type_names.dart:38]: https://github.com/mobxjs/mobx.dart/issues/850

**Comment by amondnet** [comment]
`dart run test --coverage=./coverage` sometimes failed, we should use `dart pub global run coverage:test_with_coverage`

**Review by pavanpodila — APPROVED** [review]

**Pull request merged**